hub/api/dataset.py

```diff
@@ -27,6 +30,9 @@ def __init__(
     ):
         """Initialize a new or existing dataset.
 
+        Note:
+            Be sure to keep `DATASET_ATTRIBUTES` up-to-date when changing this class.
```

Comment:
I actually meant for the user to know that he/she can't use DATASET_ATTRIBUTES since it will be overwritten by dataset attributes.

Reply:
OH, I see. On it.